server/gitrest/src/utils/helpers.ts

```diff
@@ -30,3 +31,54 @@ export function getExternalWriterParams(params: string | undefined): IExternalWr
     }
     return undefined;
 }
+
+export async function exists(fileSystemManager: IFileSystemManager, fileOrDirectoryPath: PathLike): Promise<boolean> {
+    try {
+        await fileSystemManager.stat(fileOrDirectoryPath);
+        return true;
+    } catch (error) {
+        if (error?.code === "ENOENT") {
+            // File/Directory does not exist.
+            return false;
+        }
+        throw error;
+    }
+}
+
+const latestFullSummaryFilename = "latestFullSummary";
+const getLatestFullSummaryFilePath = (dir: string) => `${dir}/${latestFullSummaryFilename}`;
+
+export async function persistLatestFullSummaryInStorage(
+    fileSystemManager: IFileSystemManager,
+    storageDirectoryPath: string,
+    latestFullSummary: IWholeFlatSummary,
+): Promise<void> {
+    const directoryExists = await exists(fileSystemManager, storageDirectoryPath);
+    if (!directoryExists) {
+        await fileSystemManager.mkdir(storageDirectoryPath);
```

Comment:
Could `storageDirectory` be something like `a/b/c`? If `a` or `b` do not exist in the file system, `mkdir` will fail unless we pass a `{ recursive: true }` parameter.

Reply:
Good question! I figured in the current usecase, it would always be `a/b/c` where `a` is "fluid", `b` is "tenantId", `c` is document id, and in order to reach this point, `a/b` would have to exist for a repo to exist. You're totally right though, that this function should take into account the potential for a more complex or alternate path. It wouldn't hurt to add recursive: true :)